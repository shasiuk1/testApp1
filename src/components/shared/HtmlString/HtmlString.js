/** @format */

import React from 'react';
import types from 'SRC/types';

const HtmlString = ({ children }) => (
  <span dangerouslySetInnerHTML={{ __html: children }} />
);

HtmlString.propTypes = {
  children: types.children,
};

HtmlString.defaultProps = {
  children: '',
};

export default HtmlString;
