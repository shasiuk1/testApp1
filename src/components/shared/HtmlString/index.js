/** @format */

import HtmlString from './HtmlString';

export default HtmlString;
