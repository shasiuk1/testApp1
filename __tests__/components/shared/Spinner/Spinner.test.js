/** @format */

import { shallow } from 'enzyme/build';
import React from 'react';
import HtmlString from 'SRC/components/shared/HtmlString';
import toJson from 'enzyme-to-json';

const testString = `Test String with <p>html</p>`;

describe('test HtmlString', () => {
  it('component is defined', () => {
    const wrapper = shallow(<HtmlString />);
    expect(wrapper.find(HtmlString)).toBeDefined();
  });

  it('component matches snapshot', () => {
    const wrapper = shallow(<HtmlString>{testString}</HtmlString>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
