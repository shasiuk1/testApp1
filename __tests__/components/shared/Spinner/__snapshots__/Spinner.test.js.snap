// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HtmlString component matches snapshot 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Test String with <p>html</p>",
    }
  }
/>
`;
